src/org/zaproxy/zap/extension/AllInOneNotes/ExtensionAllInOneNotes.java

```diff
@@ -0,0 +1,211 @@
+        super(NAME);
+        setI18nPrefix(PREFIX);
+    }
+
+    @Override
+    public void hook(ExtensionHook extensionHook) {
+        super.hook(extensionHook);
+
+        // As long as we're not running as a daemon
+        if (getView() != null) {
+
+            reload.addActionListener(l -> {
+                hookView.addStatusPanel(getStatusPanel());
+            });
+
+            extensionHook.getHookMenu().addToolsMenuItem(getMenuExample());
+            hookView = extensionHook.getHookView();
+            hookView.addStatusPanel(getStatusPanel());
+        }
+    }
+
+    @Override
+    public boolean canUnload() {
+        // The extension can be dynamically unloaded, all resources used/added can be freed/removed from core.
+        return true;
+    }
+
+    @Override
+    public void unload() {
+        super.unload();
+
+        // In this example it's not necessary to override the method, as there's nothing to unload
+        // manually, the components added through the class ExtensionHook (in hook(ExtensionHook))
+        // are automatically removed by the base unload() method.
+        // If you use/add other components through other methods you might need to free/remove them
+        // here (if the extension declares that can be unloaded, see above method).
+    }
+
+    private AbstractPanel getStatusPanel() {
+
+        ExtensionHistory extHist = (ExtensionHistory) org.parosproxy.paros.control.Control.getSingleton().
+                getExtensionLoader().getExtension(ExtensionHistory.NAME);
+
+
+        ArrayList<String[]> notes = new ArrayList<>();
+
+        if (extHist != null) {
+
+            int LastHistoryId = extHist.getLastHistoryId();
+
+            int i=0;
+            while (i++ < LastHistoryId) {
+                HistoryReference hr = extHist.getHistoryReference(i);
+                if (hr != null) {
+                    if (hr.hasNote()) {
+                        try {
+                            String note = hr.getHttpMessage().getNote();
+                            String[] tableRow = {String.valueOf(i), note};
+                            notes.add(tableRow);
+                        } catch (HttpMalformedHeaderException e) {
+                        } catch (DatabaseException e) {
+                            e.printStackTrace();
+                        }
+                    }
+                }
+            }
+        }
+
+
+        JXTable notesTable = new JXTable(new NotesTableModel(notes));
+        notesTable.setColumnSelectionAllowed(false);
+        notesTable.setCellSelectionEnabled(false);
+        notesTable.setRowSelectionAllowed(true);
+        notesTable.setAutoCreateRowSorter(true);
+        notesTable.setColumnControlVisible(true);
+
+        exportButton = new TableExportButton<>(notesTable);
+
+        tableSelectionModel = notesTable.getSelectionModel();
+        tableSelectionModel.addListSelectionListener(new NotesTableSelectionHandler(notesTable, extHist));
```

Comment:
Does tableSelectionModel need to be established just to add the listener? Couldn't it be done all at once?

Reply:
agreed! changed